### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by willyborankin**
1. Moved RecordSender class into package io.aiven.kafka.connect.http.recordsender package
2. Moved HttpSender into io.aiven.kafka.connect.http.sender package
   and split up HttpSender behave for 2 diff auth type with auth and without auth
3. Moved backoff policy into HttpSender

### CODE REVIEW

File: src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    protected final HttpSinkConfig config;
+
+    protected interface OnResponseHandler {
+
+        void onResponse(final HttpResponse<String> response) throws IOException;
+
+    }
+
+    protected final OnResponseHandler onHttpErrorResponseHandler = response -> {
+        if (response.statusCode() >= 400) {
```
Comment (HelenMel) [inline src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java:57]: I would say, that  anythin except 2XX should look suspicious. I'm not sure that connector users expect, for example 1XX responses

File: src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    /**
+     * Sends a HTTP body using {@code httpSender}, respecting the configured retry policy.
+     *
+     * @return whether the sending was successful.
+     */
+    protected void sendWithRetries(final HttpRequest request,
+                                   final OnResponseHandler onHttpResponseHandler) {
+        int remainRetries = config.maxRetries();
+        while (true) {
```
Comment (HelenMel) [inline src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java:88]: I wonder why not ```while(remainRetries > 0) ```

**Comment by HelenMel** [comment]
Looks good! 2 small comments

File: src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java (hunk 1 above)
Comment (willyborankin) [reply src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java:57]: Agree. Will change

File: src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java (hunk 2 above)
Comment (willyborankin) [reply src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java:88]: True

File: src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java (hunk 1 above)
Comment (willyborankin) [reply src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java:57]: 3xx is not covered. I did see anything in HttpClient configuration related to redirect

Comment (willyborankin) [reply src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java:57]: So HTTP status >= 400 is client errors and server errors, lets keep 400 so far but log all others statuses as warn

File: src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    protected interface OnResponseHandler {
+
+        void onResponse(final HttpResponse<String> response) throws IOException;
+
+    }
+
+    protected final OnResponseHandler onHttpErrorResponseHandler = response -> {
+        if (response.statusCode() >= 400) {
+            log.warn("Got unexpected HTTP status code: {}", response.statusCode());
```
Comment (HelenMel) [inline src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java:58]: Should this be under if (response.statusCode < 200 || response.statusCode > 299) ???

**Review by HelenMel — APPROVED** [review]

**Pull request merged**